### PR TITLE
Implement ping sending in the Python bindings

### PR DESCRIPTION
**Pull request by mdboom**

### CODE REVIEW

File: glean-core/android/src/main/java/mozilla/telemetry/glean/net/HttpURLConnectionUploader.kt [hunk 1]
```diff
@@ -90,7 +90,7 @@ class HttpURLConnectionUploader : PingUploader {
                     // Known other errors:
                     // 500 - internal error
 
-                    // For all other errors we log a warning an try again at a later time.
```
Comment (travis79) [inline glean-core/android/src/main/java/mozilla/telemetry/glean/net/HttpURLConnectionUploader.kt]: If you keep finding and fixing a couple of these with each python PR, you the Kotlin is going to be quite tidy :D

File: glean-core/python/glean/glean.py [hunk 2]
```diff
@@ -50,12 +55,16 @@ class Glean:
...
+    # The ping uploader implementation
+    _ping_uploader: BaseUploader = HttpClientUploader()
+
     @classmethod
     def initialize(
         cls,
         configuration: Optional[Configuration] = None,
         application_id: Optional[str] = None,
         data_dir: Optional[Path] = None,
+        ping_uploader: Optional[PingUploader] = None,
```
Comment (Dexterp37) [inline glean-core/python/glean/glean.py]: Let's be consistent with what [Kotlin does](https://github.com/mozilla/glean/blob/0b2f2ceea6171a3c6551345f0075bf51031a053a/glean-core/android/src/main/java/mozilla/telemetry/glean/config/Configuration.kt#L94) and add that as part of the configuration object.

File: glean-core/python/glean/glean.py [hunk 3]
```diff
@@ -85,12 +97,12 @@ def initialize(
         if data_dir is None:
             data_dir = Path(tempfile.TemporaryDirectory().name)
 
+        if ping_uploader is None:
+            ping_uploader = HttpClientUploader()
```
Comment (Dexterp37) [inline glean-core/python/glean/glean.py]: Isn't this already set on line 59?

File: glean-core/python/glean/glean.py [hunk 4]
```diff
@@ -219,5 +240,60 @@ def get_data_dir(cls) -> Path:
...
+    def test_collect(cls, ping: "PingType") -> str:
+        """
+        Collect a ping and return as a string.
+        """
+        return _ffi.ffi_decode_string(
+            _ffi.lib.glean_ping_collect(cls._handle, ping._handle)
+        )
+
+    @classmethod
+    def send_pings(cls, pings: List["PingType"]):
```
Comment (Dexterp37) [inline glean-core/python/glean/glean.py]: Is there any way we could flag/make this private or only accessible from within this package?

File: glean-core/python/glean/glean.py [hunk 5]
```diff
@@ -219,5 +240,60 @@ def get_data_dir(cls) -> Path:
...
+        Args:
+            pings (list of PingType): List of pings to send.
+        """
+        ping_names = [ping.name for ping in pings]
+
+        cls.send_pings_by_name(ping_names)
+
+    @classmethod
+    @Dispatcher.task
+    def send_pings_by_name(cls, ping_names: List[str]):
```
Comment (Dexterp37) [inline glean-core/python/glean/glean.py]: As above, any chance we could make this visible only from within this package?

File: glean-core/python/glean/glean.py (hunk 4 above)
Comment (mdboom) [reply glean-core/python/glean/glean.py]: You can't really make anything private in Python, but we can flag it with a `_` prefix.  I missed these were internal in the Kotlin implementation.

**Pull request merged**